src/caret_analyze/architecture/architecture.py

```diff
@@ -305,6 +306,46 @@ def _verify(nodes: Collection[NodeStruct]) -> None:
                      f'callback_type: {uniqueness_violated[0]}'
                      f'period_ns: {uniqueness_violated[1]}'))
 
+    def assign_message_context(self, node_name: str, context_type: str,
+                               subscribe_topic_name: str, publish_topic_name: str):
+        node: NodeStruct =\
+            Util.find_one(lambda x: x.node_name == node_name, self._nodes)
+
+        if publish_topic_name not in node.publish_topic_names:
+            raise ItemNotFoundError('{pub_topic_name} is not found in {node_name}')
+
+        if subscribe_topic_name not in node.subscribe_topic_names:
+            raise ItemNotFoundError('{sub_topic_name} is not found in {node_name}')
+
```

Comment:
It seems that these implementations are redundant.

```suggestion

```

Reply:
@hsgwa 
This is an implementation that returns errors when trying to assign a topic that does not exist in `Architecture::NodeStruct`.

From that point of view, this error handling needs to remain but should be replaced with `InvalidArgumentError`.